### PR TITLE
Makes patting someone down always damage you if your gloves aren't fire resistant

### PR DESCRIPTION
**Pull request by MatheusIsNotAvailable**
[BALANCE] [WIKI]

## About the PR 
Makes patting someone who is on fire check for your glove's heatprot, if it is equal to or higher than 7 you are safe, and the pats are more effective. the only gloves that will be safe are black gloves, swat gloves and some matsci ones depending on how you make them. also removes the 50% chance for it to not damage you at all.
## Why's this needed? 
Helps make different gloves more unique, more things for the player to think about when making a choice is good
## Changelog

```changelog
(u)Arthur Holiday
(+)If your gloves have atleast 7% heat resistance, patting out people on fire won't damage you and you will be 2x more effective at it
```

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
My concern is that this would increase player apathy when it comes to people on fire, Id much prefer to see fire resistant gloves just be far more effective at patting out fires

File: code/mob/melee_attack_procs.dm [hunk 1]
```diff
@@ -34,13 +34,21 @@
...
 	src.visible_message("<span class='notice'>[src] pats down [M] wildly, trying to put out the fire!</span>")
+	var/glove_protection = FALSE
 
-	if (prob(50))
+	if (ishuman(src)) // glove heat resistance check
+		var/mob/living/carbon/human/H = src
+		var/obj/item/clothing/gloves/G = H.gloves
+		if (G && G.hasProperty("heatprot"))
+			if (G.getProperty("heatprot") >= 7)
+				glove_protection = 1
```
Comment (BatElite) [inline code/mob/melee_attack_procs.dm:44]: Why use a bool if you can put the stuff below this here instead?

**Comment by Yellow-Mushroom** [comment]
I mean most people pat down others regardless of the damage they get so I doubt this will have any effect :/

**Comment by MatheusIsNotAvailable** [comment]
Think I will also make it more effective if you have the heat resistant gloves then

File: code/mob/melee_attack_procs.dm (hunk 1 above)
Comment (MatheusIsNotAvailable) [reply code/mob/melee_attack_procs.dm]: Changed that a commit

**Comment by penny64** [comment]
What is the gameplay purpose of only some random gloves that don't communicate their purpose having a special feature?

Look at insulated gloves, those are insulated, so we put insulated in the name.
I knew what they were for the moment I walked in the door.

Black gloves and swat gloves don't damage you when you pat people on fire? What?
And then it just disincentivizes players patting each other down?
This is bad game design

**Comment by MatheusIsNotAvailable** [comment]
Black Gloves's description says it is fire resistant and it is also shown in it's tooltips that it has more fire resistance than other gloves. This doesn't punish players all that much even, just gives them an option to do something more safely, is taking 1-2 burn damage really enough to make people not help eachother?

**Comment by ThatFiveGuys** [comment]
> Black Gloves's description says it is fire resistant and it is also shown in it's tooltips that it has more fire resistance than other gloves. This doesn't punish players all that much even, just gives them an option to do something more safely, is taking 1-2 burn damage really enough to make people not help eachother?

People can be very selfish.

**Comment by zjdtmkhzt** [comment]
I don't see the problem, people already take damage from patting, if it discouraged people from helping, it would already do so.
It just makes sense to have this be tied into your gloves fire resistance.

I dislike the RNG aspect that this has right now as well, just make it always do 1 damage if your gloves do not help and have fire resistant gloves block the damage. That makes both sense and does not punish helpers more than they are already punished right now.

**Pull request merged**